### PR TITLE
Fix Loderunner in command mode fails when multiple servers are specified

### PR DESCRIPTION
**Pull request by gortegaMS**
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

- Fix Loderunner in command mode fails when multiple servers are specified 
- Added Integration Test

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes  https://github.com/retaildevcrews/wcnp/issues/608

### CODE REVIEW

File: src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs [hunk 1]
```diff
@@ -220,7 +229,9 @@ private async Task<(bool result, string fieldName, string conditionalValue)> Val
         private async Task<int> TryParseProcessOutputAndGetAPIListeningPort(List<string> outputList, int timeBetweenTriesMs = 1000, int maxRetries = 20)
         {
             int portNumber = 0;
-            for (int i = 1; i <= maxRetries; i++)
+
+            var taskSource = new CancellationTokenSource();
+            await Common.RunAndRetry(maxRetries, timeBetweenTriesMs, taskSource, async (int attemptCount) =>
             {
                 await Task.Delay(timeBetweenTriesMs).ConfigureAwait(false); // Log Interval is 5 secs.
```
Comment (aflinchb) [inline src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs:237]: Is this Task.Delay still needed here? I see there is a Task.Delay in Common.RunAndRetry as well.

File: src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs [hunk 2]
```diff
@@ -256,22 +267,24 @@ private async Task<int> TryParseProcessOutputAndGetAPIListeningPort(List<string>
         private async Task<string> TryParseProcessOutputAndGetClientStatusId(List<string> outputList, int timeBetweenTriesMs = 500, int maxRetries = 10)
         {
             string clientStatusId = null;
-            for (int i = 1; i <= maxRetries; i++)
+            var taskSource = new CancellationTokenSource();
+
+            await Common.RunAndRetry(maxRetries, timeBetweenTriesMs, taskSource, async (int attemptCount) =>
             {
                 await Task.Delay(timeBetweenTriesMs).ConfigureAwait(false); // Log Interval is 5 secs.
```
Comment (aflinchb) [inline src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs:275]: Same question here regarding the Task.Delay

File: src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs (hunk 1 above)
Comment (gortegaMS) [reply src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs:237]: No need for it, will remove

File: src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs (hunk 2 above)
Comment (gortegaMS) [reply src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs:275]: No need for it, will remove

File: src/LodeRunner.Core/SystemConstants.cs [hunk 3]
```diff
@@ -133,5 +133,15 @@ public class SystemConstants
         /// The default API web host port.
         /// </summary>
         public const int DefaultApiWebHostPort = 8080;
+
+        /// <summary>
+        /// The integration clients URI path.
+        /// </summary>
+        public const string IntegrationClientsUri = "/api/Clients";
```
Comment (aflinchb) [inline src/LodeRunner.Core/SystemConstants.cs:140]: I likely missed it, but is the IntegrationClientsUri being used?

**Review by aflinchb — APPROVED** [review]

File: src/LodeRunner.Core/SystemConstants.cs (hunk 3 above)
Comment (gortegaMS) [reply src/LodeRunner.Core/SystemConstants.cs:140]: Removed reference and constant, and updated code to use an existing constant under LodeRunner.API.SystemConstants

**Pull request merged**